source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/products/modal.tsx

```diff
@@ -6,28 +6,41 @@ import { withTranslations, i18nProps } from '@lib/i18n';
 
 class ProductModal extends React.Component<i18nProps> {
 
+  state = {
+    isModalOpen: false
+  }
+
   render() {
 
     const { t } = this.props;
+    const { isModalOpen } = this.state;
 
     const trigger = (
-      <button className='ui button'>
+      <button
+        className='ui button'
+        onClick={() => this.setState({ isModalOpen: true })}
```

Comment:
`() => this.setState({ isModalOpen: true })` is kinda repeated 3 times

do you have thoughts on:
```ts
import { withTemplateHelpers, Toggle } from 'react-template-helpers';

@withTemplateHelpers
class ProductModal ... {
  toggle: Toggle;

  // ...

  const toggleModal = toggle('isModalOpen');

  // ...

  onClose={toggleModal}
}
```

(other than being a total of 1 extra line of code)

Reply:
I'm like 50 / 50 but since we are using it in other places I will use it